### PR TITLE
Example: Use RNNs to answer questions from bAbi

### PR DESCRIPTION
**Pull request by Smerity**
Trains two recurrent neural networks based upon a story and a question.
The resulting merged vector is then queried to answer a range of bAbI tasks.

The results are comparable to those for an LSTM model provided in Weston et al.:
"Towards AI-Complete Question Answering: A Set of Prerequisite Toy Tasks"
http://arxiv.org/abs/1502.05698

With default word, sentence, and query vector sizes, the GRU model achieves:
  - 52.1% test accuracy on QA1 in 20 epochs (2 seconds per epoch on CPU)
  - 37.0% test accuracy on QA2 in 20 epochs (16 seconds per epoch on CPU)

In comparison, the Facebook paper achieves 50% and 20% for the LSTM baseline.
Whilst this is not a task where the RNNs is expected to win, I still feel it provides a good demonstration.

The task utilises multiple features from Keras that do not yet have good examples, including embedding layers, masks, and merging.

### CODE REVIEW

**Comment by fchollet** [comment]
Cool stuff! Merged.
